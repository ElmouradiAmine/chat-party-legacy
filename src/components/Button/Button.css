.button {
  text-align: center;
  border-radius: 3px;
  padding: 1rem 1.2rem;
  border: none;
  color: white;
  text-transform: uppercase;
  font-weight: 700;
  box-shadow: 0px 2px 1px rgba(0, 0, 0, 0.1);
  transition: all 0.2s;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 1.6rem;
  font-family: 'Poppins', sans-serif;
}

.button:hover {
  filter: brightness(110%);
  transform: translateY(-3px) scaleX(1.05);
}

.button:active {
  filter: brightness(90%);
  transform: translateY(-1px) scaleX(1.05);
}

.button--green {
  background-color: var(--color-green);
}

.button--red {
  background-color: var(--color-red);
}

.button--yellow {
  background-color: var(--color-yellow);
}

.button:disabled {
  background-color: grey;
}
