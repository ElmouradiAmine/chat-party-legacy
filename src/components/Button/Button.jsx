import React from 'react';
import PropTypes from 'prop-types';

import './Button.css';
import Spinner from '../Spinner/Spinner';

function Button({
  className, label, color, onClick, disabled,
}) {
  return (
    <button
      aria-label={label}
      type="button"
      className={`button button--${color} ${className}`}
      onClick={onClick}
      disabled={disabled}
    >
      {!disabled ? `${label}` : <Spinner /> }
    </button>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  label: PropTypes.string,
  onClick: PropTypes.func,
  color: PropTypes.string,
  disabled: PropTypes.bool,
};

Button.defaultProps = {
  className: '',
  label: 'Button',
  onClick: null,
  color: 'green',
  disabled: false,
};

export default Button;
