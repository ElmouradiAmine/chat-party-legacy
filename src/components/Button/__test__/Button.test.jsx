import React from 'react';
import { cleanup, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import Button from '../Button';

afterEach(cleanup);
describe('Button component', () => {
  const {
    className, label, onClick, color, disabled,
  } = {
    className: 'className',
    label: 'button',
    onClick: jest.fn(),
    color: 'red',
    disabled: false,
  };

  test('renders correctly', () => {
    render(<Button />);
  });

  test('renders correctly with props', () => {
    const { container, getByText } = render(
      <Button
        label={label}
        className={className}
        color={color}
        onClick={onClick}
        disabled={disabled}
      />,
    );
    expect(getByText(label)).toBeTruthy();
    expect(container.firstChild.classList).toContain(className);
    expect(container.firstChild.classList).toContain(`button--${color}`);
  });

  test('should get clicked', () => {
    const { container } = render(<Button onClick={onClick} />);
    userEvent.click(container.firstChild);
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Button
          label={label}
          className={className}
          color={color}
          onClick={onClick}
          disabled={disabled}
        />,
      );

    expect(tree).toMatchSnapshot();
  });
});
