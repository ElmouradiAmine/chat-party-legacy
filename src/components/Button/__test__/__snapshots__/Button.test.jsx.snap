// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<button
  aria-label="button"
  className="button button--red className"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  button
</button>
`;
